### PR TITLE
Remove get-by-title from NPM package API

### PR DESCRIPTION
**Pull request by ericcornelissen**
**Issue:** Relates to #2260 and #4797
**Alexa rank:** n/a

### Checklist
  - [n/a] I updated the JSON data in `_data/simple-icons.json`
  - [n/a] I optimized the icon with SVGO or SVGOMG
  - [n/a] The SVG `viewbox` is `0 0 24 24`

### Description

Based on the discussion in #2260, with the intent to allow for brands with the same name (read `"title"`) to be included in the project, this removes the ability to `simpleIcons.get()` an icon from the NPM package based on the title.

This does beg the question whether we should convert back to the old API:

```js
const simpleIcons = require('simple-icons');

console.log(simpleIcons['simpleicons']); // Use the filename of the SVG

/*
{
    title: 'Simple Icons',
    slug: 'simpleicons',
    hex: '111111',
    source: 'https://simpleicons.org/',
    svg: '<svg role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">...</svg>',
    path: 'M12 12v-1.5c-2.484 ...'
}
*/
```

I would be okay with that, but I would also like to explore the possibility of making the `simpleIcons` object both [enumerable](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in) and [iterable](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) and am unsure if these three things can exist at the same time.

### CODE REVIEW

**Comment by PeterShaggyNoble** [comment]
Thought: to minimise the number of times a user may potentially need to check [the look-up table](https://github.com/simple-icons/simple-icons/pull/5002), I wonder if this could be the excuse we're looking for to expose `titleToSlug` in the package and, thus, to the website. It would allow users to continue using the `title` as much as they like. For example:

```
let slug=titleToSlug(title);
simpleIcons.get(slug);
```

**Comment by ericcornelissen** [comment]
> Thought: to minimise the number of times a user may potentially need to check [the look-up table](https://github.com/simple-icons/simple-icons/pull/5002), I wonder if this could be the excuse we're looking for to expose `titleToSlug` in the package and, thus, to the website. It would allow users to continue using the `title` as much as they like. For example:
> 
> ```
> let slug=titleToSlug(title);
> simpleIcons.get(slug);
> ```

I'd personally not be a fan of that as it doesn't work 100% of the time, what does it even return for brands that have a custom slug and how does it know that?

I think the best we can do is improve the experience of the look-up table, e.g. with a website or a CLI script made available through NPM. But those seem more like above-and-beyond features to me.

**Comment by PeterShaggyNoble** [comment]
> I'd personally not be a fan of that as it doesn't work 100% of the time

Yeah, it was just a though, trying to find _some_ excuse to add `titleToSlug` to the package so we don't have to keep it manually synced on the website! 😆

> what does it even return for brands that have a custom slug and how does it know that?

It would return the one without the custom slug which, of course, could lead to confusion if people don't get the icon they're expecting. At that stage, though, they could go to the lookup table. The thought was to try to reduce the amount of times people have to look up the slug if they're using a lot of icons but that's probably the lesser of 2 evils compared to them not getting the icon they're expecting.

> I think the best we can do is improve the experience of the look-up table, e.g. with a website

Displaying it somewhere on the new website would be useful, perhaps as part of [#43](https://github.com/simple-icons/simple-icons-website/issues/43) if it gets implemented.

**Comment by ericcornelissen** [comment]
> > I'd personally not be a fan of that as it doesn't work 100% of the time
> 
> Yeah, it was just a though, trying to find _some_ excuse to add `titleToSlug` to the package so we don't have to keep it manually synced on the website! laughing

[The `titleToSlug` function](https://github.com/simple-icons/simple-icons/blob/4887643364e4ddda1638460bb1297df403addb05/scripts/utils.js#L17-L37) and  [the `normalizeSearchTerm` function](https://github.com/simple-icons/simple-icons-website/blob/45309e58a3a2075dc869262d1983482d976104d6/public/scripts/utils.js#L33-L46) are different anyway, so unless we split the overlapping parts into a third function here, exposing `titleToSlug` doesn't help the website I'm afraid.


> > what does it even return for brands that have a custom slug and how does it know that?
> 
> It would return the one without the custom slug which, of course, could lead to confusion if people don't get the icon they're expecting. At that stage, though, they could go to the lookup table. The thought was to try to reduce the amount of times people have to look up the slug if they're using a lot of icons but that's probably the lesser of 2 evils compared to them not getting the icon they're expecting.

Fair enough. I would definitely go crazy if that function gives me the wrong name and hence icon though, hence my confusion as to what should be returned. Your suggestion actually makes a lot of sense if you think about it, but I don't think it does as an end-user. That said, if others agree with you I'm happy to expose `titleToSlug` for this purpose :slightly_smiling_face: 


> > I think the best we can do is improve the experience of the look-up table, e.g. with a website
> 
> Displaying it somewhere on the new website would be useful, perhaps as part of [#43](https://github.com/simple-icons/simple-icons-website/issues/43) if it gets implemented.

Definitely :+1:

**Comment by PeterShaggyNoble** [comment]
Couple of conflicts to be resolved in this one, @ericcornelissen.

File: scripts/build/package.js [hunk 1]
```diff
@@ -34,11 +34,7 @@ function escape(value) {
   return value.replace(/(?<!\\)'/g, "\\'");
 }
 function iconToKeyValue(icon) {
-  let iconName = escape(icon.title);
-  if (icon.slug !== titleToSlug(icon.title)) {
-    iconName = icon.slug;
-  }
-
+  const iconName = icon.slug;
   return `'${iconName}':${iconToObject(icon)}`;
```
Comment (mondeja) [inline scripts/build/package.js:38]: Minor suggestion:
```suggestion
  return `'${icon.slug}':${iconToObject(icon)}`;
```

**Review by mondeja — APPROVED** [review]
The next release will be faster, LGTM!
I've a minor suggestion but not important at all, good work :+1:

**Comment by PeterShaggyNoble** [comment]
Merging this in based on @mondeja's approval above. Nice work, @ericcornelissen 👍🏻

**Pull request merged**